### PR TITLE
installer: add optional Azure Mag e2e job

### PR DESCRIPTION
**Pull request by r4f4**
Signed-off-by: Rafael Fonseca <r4f4rfs@gmail.com>

### CODE REVIEW

File: ci-operator/step-registry/openshift/e2e/azuremag/OWNERS [hunk 1]
```diff
@@ -0,0 +1,7 @@
+approvers:
+- staebler
+- rna-afk
+- patrickdillon
+- mtnbikenc
```
Comment (patrickdillon) [inline ci-operator/step-registry/openshift/e2e/azuremag/OWNERS:5]:
```suggestion
- kirankt
```

File: ci-operator/step-registry/openshift/e2e/azuremag/openshift-e2e-azuremag-workflow.metadata.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+	"path": "openshift/e2e/azuremag/openshift-e2e-azuremag-workflow.yaml",
+	"owners": {
+		"approvers": [
+			"staebler",
+			"rna-afk",
+			"patrickdillon",
+			"mtnbikenc",
```
Comment (patrickdillon) [inline ci-operator/step-registry/openshift/e2e/azuremag/openshift-e2e-azuremag-workflow.metadata.json:8]:
```suggestion
			"kirankt",
```

**Review by patrickdillon — COMMENTED** [review]
looks like a good start. some quick comments hopefully a rehearsal will run.

File: ci-operator/config/openshift-priv/installer/openshift-priv-installer-master.yaml [hunk 3]
```diff
@@ -447,6 +447,11 @@ tests:
   steps:
     cluster_profile: azurestack
     workflow: openshift-e2e-azurestack-upi
+- as: e2e-azuremag
+  optional: true
+  steps:
+    cluster_profile: azuremag
+    workflow: openshift-e2e-azurestack
```
[the remark this answers is not in the document]
Comment (r4f4) [reply ci-operator/config/openshift-priv/installer/openshift-priv-installer-master.yaml]: Oops, fixed.

File: ci-operator/jobs/openshift-priv/installer/openshift-priv-installer-master-presubmits.yaml [hunk 4]
```diff
@@ -1249,6 +1249,89 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-azure-upi,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
[the remark this answers is not in the document]
Comment (r4f4) [reply ci-operator/jobs/openshift-priv/installer/openshift-priv-installer-master-presubmits.yaml:1253]: Do you know how one sets that to `false` when this file is auto-generated?

[the remark this answers is not in the document]
Comment (patrickdillon) [reply ci-operator/jobs/openshift-priv/installer/openshift-priv-installer-master-presubmits.yaml:1253]: It is ok to change the value to false after the file is generated. When files are subsequently generated, the false value will not be overwritten.

**Comment by patrickdillon** [comment]
It looks like the config and the job are in the wrong org `openshift-priv` -> `openshift`

File: ci-operator/jobs/openshift-priv/installer/openshift-priv-installer-master-presubmits.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (r4f4) [reply ci-operator/jobs/openshift-priv/installer/openshift-priv-installer-master-presubmits.yaml]: Thanks. Fixed.

**Comment by r4f4** [comment]
I've fixed the job/config location.

**Comment by patrickdillon** [comment]
In general this looks good to me. I wish we could get a rehearsal, but that doesn't seem possible. I don't have full approval rights so it will be good to get a second pair of eyes on this if @staebler takes a look.

/lgtm

**Comment by r4f4** [comment]
The `azuremag` job passed though https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/26725/rehearse-26725-pull-ci-openshift-installer-master-e2e-azuremag/1499501161818886144

**Comment by patrickdillon** [comment]
@Amoghrd do you know if there are any issues if we piggy back a job based off the work you did to setup MAG CI? I want to make sure the subscription is not reserved for QE, or anything like that. I am also asking infra services to make sure this is ok.

If we are moving forward, I noticed this rehearsal is using the QE basedomain so we should change that.

**Comment by Amoghrd** [comment]
@patrickdillon I am not sure if the subscription is only for QE. Maybe @jianlinliu can provide more insight on this

**Comment by jianlinliu** [comment]
@patrickdillon It is okay to update the job to use the MAG CI account instead of QE's MAG account, can QE access the MAG CI account and the the credential saved in vault?

**Review by staebler — COMMENTED** [review]
This looks good to me, outside of the account concerns.

**Comment by jianlinliu** [comment]
> is looks good to me, outside of the account concerns.

We are going to create QE's dedicated profile to resolve this concerns.

**Pull request closed without merging**